includes/definitions/discovery/bintec-beip-plus.yaml

```diff
@@ -0,0 +1,26 @@
+# Some values are hardcoded because snmpbulkwalk is not supported
+modules:
+    os:
+        version: BIANCA-BRICK-MIB::biboAdmSWVersion.0
+        version_regex: '/(?<version>[V]{1}[.0-9]+)/'
+        serial: BIANCA-BRICK-MIB::biboABrdSerialNo.0.0.0
+        hardware: 
+            - BIANCA-BRICK-MIB::biboABrdPartNo.0.0.0
+            - BIANCA-BRICK-MIB::biboABrdHWRelease.0.0.0
+        hardware_template: '{{ BIANCA-BRICK-MIB::biboABrdPartNo.0.0.0 }} {{ BIANCA-BRICK-MIB::biboABrdHWRelease.0.0.0 }}'
+    mempools:
+        data:
+            -
+#                total: BIANCA-BRICK-MIBRES-MIB::memoryTotal.2.1
+#                used: BIANCA-BRICK-MIBRES-MIB::memoryInuse.2.1
```

Comment:
Any reason you used the numOID here ? Textual form would be easier to read I think.

Reply:
Kind of, yes :sweat_smile: At that time, I could not get the commented version using the textual representation to work. I did some debugging and found the reason and also was able to work around it:

https://github.com/librenms/librenms/blob/ae2b761e940b0cfc0b66badc6b6ada99f2e16f5f/LibreNMS/OS/Traits/YamlMempoolsDiscovery.php#L93

Apparently, when using the textual representation with a static index appended, the above linked method is not capable of getting the data from the array of received SNMP data. The method e.g. receives
```php
$field = "used" // To get the amount of used memory
$index = "2.1" // The correct index I was statically appending
$yaml = array (
  "total" => "BIANCA-BRICK-MIBRES-MIB::memoryTotal.2.1"
  "used" => "BIANCA-BRICK-MIBRES-MIB::memoryInuse.2.1"
);
```
Now, in the next line:
https://github.com/librenms/librenms/blob/ae2b761e940b0cfc0b66badc6b6ada99f2e16f5f/LibreNMS/OS/Traits/YamlMempoolsDiscovery.php#L95
the requested OID is fetched from the array. After that, the OID is used to fetch the requested data from the `$this->mempools` array:
https://github.com/librenms/librenms/blob/ae2b761e940b0cfc0b66badc6b6ada99f2e16f5f/LibreNMS/OS/Traits/YamlMempoolsDiscovery.php#L96
But since the fetched OID already has the index statically appended, the above lookup fails. BTW, the `$this->mempools` array looks like this:
```php
array(
"2.1" => array (
    "BIANCA-BRICK-MIBRES-MIB::memoryTotal" => someVal,
    ...
  )
);
```
which IMO correctly represents the data received by walking over the SNMP table. The `getData` function just is not capable to work with OIDs in textual form while also appending a static index.

Since I could not find documentation that states such configuration is supported/allowed, I don't consider this a bug. I was just expecting this to work, as I could use such OIDs in different places of the YAML config without issues. Maybe you are thinking differently about it? At the end, I could work around this by configuring it to walk over the entire `memoryTable` and skipping all entries that were not of interest.